python/tank/platform/engine.py

```diff
@@ -1276,6 +1278,33 @@ def __debug_track_qt_widget(self, widget):
     ##########################################################################################
     # private and protected methods
 
+    def _emit_event(self, event):
+        """
+        Called by the engine whenever an event is to be emitted to child
+        apps of this engine.
+
+        .. note:: Events will be emitted and child apps notified immediately.
+
+        .. warning:: Some event types might be triggered quite frequently. Apps
+                     that react to events should do so in a way that is aware of
+                     the potential performance impact of their actions.
+
+        :param event: The event object that will be emitted.
+        :type event:  :class:`~sgtk.platform.events.EngineEvent`
+        """
+        self.log_debug("Emitting event: %s" % event)
+        for app_instance_name, app in self.__applications.iteritems():
```

Comment:
I think we should do an `isinstance` check here to make sure that the object sent in via the event parameter derives from `EngineEvent`.